drivers/hmis/spec/models/hmis/hmis_data_cleanup_util_spec.rb

```diff
@@ -306,4 +309,84 @@ def expect_leaves_non_hmis_data_alone(&block)
       end
     end
   end
+
+  context 'with household ids duplicated across projects' do
+    before(:each) do
+      p2 = create(:hmis_hud_project, data_source: hmis_ds, organization: o1)
+      e1.update(household_id: 'ABCDEF', project: p1)
+      e2.update(household_id: 'ABCDEF', project: p2)
+      e3.update(household_id: 'ABCDEF', project: p2)
+    end
+
+    it 'reassigns household IDs correctly' do
+      expect { HmisDataCleanup::Util.reassign_duplicate_household_ids! }.to(
+        [
+          change { Hmis::Hud::Enrollment.find(e1.id).household_id },
+          change { Hmis::Hud::Enrollment.find(e2.id).household_id },
+          change { Hmis::Hud::Enrollment.find(e3.id).household_id },
+        ].reduce(&:and),
+      )
+
+      [e1, e2, e3].map(&:reload)
+      expect(e1.household_id).not_to eq(e2.household_id)
+      expect(e2.household_id).to eq(e3.household_id), 'household remains intact'
+    end
+
+    it 'has no side-effects' do
+      expect_leaves_non_hmis_data_alone do
+        HmisDataCleanup::Util.reassign_duplicate_household_ids!
+      end
+    end
+  end
+
+  context 'with duplicate deleted EnrollmentIDs' do
+    let(:dup_key) { 'ABCDEF' }
+    before(:each) do
+      e1.update(EnrollmentID: dup_key, DateDeleted: 1.week.ago, project: p1)
+      e2.update(EnrollmentID: dup_key, DateDeleted: nil, project: p1)
+      e3.update(EnrollmentID: dup_key, DateDeleted: 1.week.ago, project: p1)
+
+      # duplicate in a different project
+      p2 = create(:hmis_hud_project, data_source: hmis_ds, organization: o1)
+      e4.update(EnrollmentID: dup_key, DateDeleted: 1.week.ago, project: p2)
```

Comment:
probably want to get exception if this fails
```suggestion
      e1.update!(EnrollmentID: dup_key, DateDeleted: 1.week.ago, project: p1)
      e2.update!(EnrollmentID: dup_key, DateDeleted: nil, project: p1)
      e3.update!(EnrollmentID: dup_key, DateDeleted: 1.week.ago, project: p1)

      # duplicate in a different project
      p2 = create(:hmis_hud_project, data_source: hmis_ds, organization: o1)
      e4.update!(EnrollmentID: dup_key, DateDeleted: 1.week.ago, project: p2)
```